src/util/pagination.tsx

```diff
@@ -17,18 +18,51 @@ export const paginationOptions = [
   },
 ];
 
-export const usePagination = (
-  data: MainTableRow[]
-): {
+export interface Pagination {
   pageData: MainTableRow[];
   currentPage: number;
   paginate: (pageNumber: number) => void;
   pageSize: number;
   setPageSize: (newSize: number) => void;
   totalPages: number;
-} => {
+  updateSort: (sort: string | null | undefined) => void;
+}
+
+export const usePagination = (
+  data: MainTableRow[],
+  defaultSort?: string,
+  defaultSortDirection?: SortDirection
+): Pagination => {
   const [_pageSize, _setPageSize] = useState(paginationOptions[0].value);
 
+  const [sort, setSort] = useState<string | null | undefined>(defaultSort);
+  const [sortDirection, setSortDirection] = useState(
+    defaultSortDirection ?? "ascending"
+  );
```

Comment:
We can have an initializer, and get rid of the `??` operator in the state:

```suggestion
  defaultSortDirection: SortDirection = "ascending"
): Pagination => {
  const [_pageSize, _setPageSize] = useState(paginationOptions[0].value);

  const [sort, setSort] = useState<string | null | undefined>(defaultSort);
  const [sortDirection, setSortDirection] = useState(defaultSortDirection);
```

Reply:
ah right, good idea.